magefiles/utils.go

```diff
@@ -441,8 +441,15 @@ func HandleErrorWithAlert(err error, errLevel slack.ErrorSeverityLevel) error {
 	return nil
 }
 
-func getChangedFiles() (rulesengine.Files, error) {
+func getChangedFiles(repo string) (rulesengine.Files, error) {
 	var gitDiff = sh.OutCmd("git", "diff")
+	// If the repo is infra-deployments then we check changed files in path tmp/infra-deployments
+	// else it will run locally which is from e2e-tests directory
+	if repo == "infra-deployments" {
+		// gitDiff = sh.OutCmd(fmt.Sprintf("git -C tmp/%s", repo), "diff")
+		wd, _ := os.Getwd()
+		gitDiff = sh.OutCmd("git", "-C", fmt.Sprintf("%s/tmp/%s", wd, repo), "diff")
+	}
 	output, err := gitDiff("--name-status", "main..HEAD")
```

Comment:
so we [hit an issue in infra-deployments PR](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/redhat-appstudio_infra-deployments/4351/pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-tests/1824130672202092544#1:build-log.txt%3A1460-1464) when using this git diff argument


I think this should fix it:
```suggestion
	output, err := gitDiff("--name-status", "upstream/main")
```